### PR TITLE
Introduce libvm and a resource builder within libinstancetype

### PR DESCRIPTION
**Pull request by lyarwood**
/area instancetype

**What this PR does / why we need it**:

This is another follow up series for https://github.com/kubevirt/kubevirt/pull/10456 and will remain draft until that lands.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [x] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [x] PR: The PR description is expressive enough and will help future contributors
- [x] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [x] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [x] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [x] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [x] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [x] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: tests/libvm/vm.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	vm := virtv1.NewVMReferenceFromNameWithNS("", name)
+	vm.Spec = virtv1.VirtualMachineSpec{}
+	vm.TypeMeta = k8smetav1.TypeMeta{
+		APIVersion: virtv1.GroupVersion.String(),
+		Kind:       "VirtualMachine",
+	}
+	return vm
+}
+
+func WithVMITemplateSpec(vmiOpts ...libvmi.Option) Option {
```
Comment (0xFelix) [inline tests/libvm/vm.go:37]: Take a `vmi` as parameter instead and call it with `libvmi.NewCirros()`?

**Review by 0xFelix — COMMENTED** [review]
Can generics help to reduce the amount of code for the instancetype/preference builders?

**Comment by lyarwood** [comment]
/close

I'm going to cherry-pick a few of these commits into a new PR.

**Pull request closed without merging**